src/team_lib/mhba/xccl_mhba_ib.h

```diff
@@ -6,7 +6,52 @@
 #ifndef XCCL_TEAM_LIB_MHBA_IB_H_
 #define XCCL_TEAM_LIB_MHBA_IB_H_
 #include "xccl_mhba_lib.h"
+#include <stdatomic.h>
 
+#define SQ_WQE_SHIFT 6
+#define mmio_flush_writes()                                                    \
+    asm volatile("sfence" ::: "memory") // TODO only for __x86_64__
+#define udma_to_device_barrier()                                               \
+    asm volatile("" ::: "memory") // TODO only for __x86_64__
+#define mmio_wc_start() mmio_flush_writes()
+
+#ifdef __CHECKER__
+#define __force __attribute__((force))
+#else
+#define __force
+#endif
+
+typedef void (*write64_fn_t)(void *, __be64);
+
+#define MAKE_WRITE(_NAME_, _SZ_)                                               \
+    static inline void _NAME_##_be(void *addr, __be##_SZ_ value)               \
+    {                                                                          \
+        atomic_store_explicit((_Atomic(uint##_SZ_##_t) *)addr,                 \
+                              (__force uint##_SZ_##_t)value,                   \
+                              memory_order_relaxed);                           \
+    }
+
+MAKE_WRITE(mmio_write32, 32)
```

Comment:
Unused?

Reply:
Used - it calls the define on top of it to define the needed function